java/vtgate-client/src/main/java/com/youtube/vitess/vtgate/Field.java

```diff
@@ -1,27 +1,214 @@
+    VT_ENUM(247, String.class),
+    VT_SET(248, String.class),
+    VT_TINY_BLOB(249, byte[].class),
+    VT_MEDIUM_BLOB(250, byte[].class),
+    VT_LONG_BLOB(251, byte[].class),
+    VT_BLOB(252, byte[].class),
+    VT_VAR_STRING(253, byte[].class),
+    VT_STRING(254, byte[].class),
+    VT_GEOMETRY(255, byte[].class);
+
+    public final int mysqlType;
+    public final Class javaType;
+    public final Class unsignedJavaType;
+
+    FieldType(int mysqlType, Class javaType) {
+      this.mysqlType = mysqlType;
+      this.javaType = javaType;
+      this.unsignedJavaType = javaType;
+    }
+
+    FieldType(int mysqlType, Class javaType, Class unsignedJavaType) {
+      this.mysqlType = mysqlType;
+      this.javaType = javaType;
+      this.unsignedJavaType = unsignedJavaType;
+    }
+  }
+  
   private String name;
   private FieldType type;
-
-  public Field(String name, FieldType type) {
-    this.name = name;
-    this.type = type;
+  private int flags;
+  
+  public static Field newFieldFromMysql(String name, int mysqlTypeId, int mysqlFlags) {
+    for (FieldType ft : FieldType.values()) {
+      if (ft.mysqlType == mysqlTypeId) {
+        return new Field(name, ft, mysqlFlags);
+      }
+    }
+    
+    throw new RuntimeException("Unknown MySQL type: " + mysqlTypeId);
   }
 
-  public String getName() {
-    return name;
+  public static Field newFieldForTest(FieldType fieldType, Flag flag) {
+    return new Field("dummyField", fieldType, flag.mysqlFlag);
   }
 
-  public void setName(String name) {
+  private Field(String name, FieldType type, int mysqlFlags) {
     this.name = name;
+    this.type = type;
+    this.flags = mysqlFlags;
   }
 
-  public FieldType getType() {
-    return type;
+  public Cell convertValueToCell(byte[] bytes) {
+    if ((flags & Flag.VT_UNSIGNED_FLAG.mysqlFlag) != 0) {
+      return new Cell(name, convert(bytes), type.unsignedJavaType);
+    } else {
+      return new Cell(name, convert(bytes), type.javaType);
+    }
   }
 
-  public void setType(FieldType type) {
-    this.type = type;
+  Object convert(byte[] bytes) {
+    if (bytes == null || bytes.length == 0) {
+      return null;
+    }
+    String s = null;
+    try {
+      s = new String(bytes, CharEncoding.ISO_8859_1);
+    } catch (UnsupportedEncodingException e) {
+      throw new RuntimeException(e);
+    }
+
+    switch (type) {
+      case VT_DECIMAL:
+        return new BigDecimal(s);
+      case VT_TINY:
+        return Integer.valueOf(s);
+      case VT_SHORT:
+        return Integer.valueOf(s);
+      case VT_LONG:
+        return Long.valueOf(s);
+      case VT_FLOAT:
+        return Float.valueOf(s);
+      case VT_DOUBLE:
+        return Double.valueOf(s);
+      case VT_NULL:
+        return null;
+      case VT_TIMESTAMP:
+        s = s.replace(' ', 'T');
+        return DateTime.parse(s);
+      case VT_LONGLONG:
+        // This can be an unsigned or a signed long
+        if ((flags & Flag.VT_UNSIGNED_FLAG.mysqlFlag) != 0) {
```

Comment:
Do you want to delay this change? If the client gets deployed before the server, you may get unexpected results.